core/dbt/task/runnable.py

```diff
@@ -371,6 +371,9 @@ def _mark_dependent_errors(self, node_id, result, cause):
             self._skipped_children[dep_node_id] = cause
 
     def populate_adapter_cache(self, adapter, required_schemas: Set[BaseRelation] = None):
+        if not self.args.populate_cache:
+            return
```

Comment:
There is no real difference, no. `self.args` is `Flags` which we should be using much more where we can. I think the places we use`get_flags` instead of `self.args` was just to get the click feature branch over the line to be merged.